### ISSUE TITLE
What should be the input data range

### ISSUE THREAD

**Issue by 18445864529** [ragor114/PyTorch-Frechet-Video-Distance#1]
`0~255` or `0~1` or `-1~1`?
Thanks!

**Comment by ragor114** [comment]
Hi @18445864529 , thanks for your interest in my implementation.
I believe the data range should be `0 ~ 1`.

**Comment by 18445864529** [comment]
Thank you for the prompt reply. 
For my data, when I tried `0 ~ 1`, the resulting value was 0.0149, `-1 ~ 1` was 0.125, and `0 ~ 255` was billion. None of them seems correct. 
I also tried `a=torch.rand(50,3,20,256,256); b=0.9*a; compute_fvd(a,b)`, and the result was also 0.0018, which does not seem right.
What is the probable reason?

**Comment by ragor114** [comment]
You are right this does not seem correct and does not correspond to the values I got with the implementation.
I'm sorry currently I do not know how to fix the problem.
If the problem persists I would recommend contacting the authors of the [StyleGAN-V repository](https://github.com/universome/stylegan-v) as my implementation is 90% taken from their source code and uses their pretrained model.
If you discover a solution please inform me!

**Comment by samuruph** [comment]
Yes I am also getting relly low results for that!

**Comment by samuruph** [comment]
Simply load the model like this:
def load_i3d(device='cuda'):
    """Load the I3D model from PyTorch Hub."""
    detector_url = 'https://www.dropbox.com/s/ge9e5ujwgetktms/i3d_torchscript.pt?dl=1'
    detector_kwargs = dict(rescale=False, resize=False, return_features=True) # Return raw features before the softmax layer.
    with open_url(detector_url, verbose=False) as f:
        detector = torch.jit.load(f).eval().to(device)
    return detector
See: https://github.com/universome/fvd-comparison/tree/master

**Comment by ragor114** [comment]
@samueleruffino99 I tried to implement your fix, however, I get almost the exact same values using my model load and the function you proposed (and the function used in the original repository).
You can find my implementation and some experiments in the [fix-model-load branch](https://github.com/ragor114/PyTorch-Frechet-Video-Distance/tree/fix-model-load). Maybe I did something wrong?

If you would open a Pull Request with a working fix, I would be happy to merge it!

**Comment by ryushinn** [comment]
Hi @ragor114 . In the original StyleGAN-V repo, doesn't this line indicate that the input range should be [0, 255] as uint8?

https://github.com/universome/stylegan-v/blob/3fecd69c602e1cda204357201461c0fb0a634909/src/metrics/metric_utils.py#L236

**Comment by lucasrelic99** [comment]
As this code is currently written, the model inputs should be in the range [0,255] and can be any resolution.

The raw detector model expects inputs in [-1,1] of resolution 224x224 ([source](https://github.com/google-deepmind/kinetics-i3d/blob/master/README.md#sample-data-and-preprocessing)) , however as it is currently implemented the inputs are preprocessed before being passed to the first model layer.

This can be verified by inspecting the detector's torchscript code:
```python
>>> print(detector.code)
def forward(self,
    x: Tensor,
    rescale: bool=False,
    resize: bool=False,
    return_features: bool=False) -> Tensor:
  _0 = __torch__.torch.nn.functional.___torch_mangle_158.interpolate
  if rescale:
    x1 = torch.sub(torch.mul(torch.div(x, 255), 2), 1)
    x0 = x1
  else:
    x0 = x
  if resize:
    b, c, t, h, w, = torch.size(x0)
    x3 = _0(torch.view(x0, [b, -1, h, w]), [224, 224], None, "bilinear", False, None, )
    x2 = torch.view(x3, [b, c, t, 224, 224])
  else:
    x2 = x0
  conv3d_1a_7x7 = self.conv3d_1a_7x7
  out = (conv3d_1a_7x7).forward(x2, )
...
```
This [code passes kwargs](https://github.com/ragor114/PyTorch-Frechet-Video-Distance/blob/431844ef10417f661dbe47832831ab0558acb340/fvd_metric/fvd.py#L140) `rescale=True, resize=True`, thus it is clear that the inputs should be in range [0,255] and can be any resolution.|

All this being said, I am still getting quite high FVD values (on the order of 100s) with this input range. So it's possible there could be another issue
